src/index.ts

```diff
@@ -142,15 +142,17 @@ class ServerlessCustomDomain {
         await this.getDomainInfo();
 
         if (autoDomain === true) {
-            const atLeastOneDoesNotExist = this.domains.some((domain) => !domain.domainInfo);
-            if (atLeastOneDoesNotExist === true) {
-                const waitFor =
-                    parseInt(this.serverless.service.custom.customDomain.autoDomainWaitFor, 10)
-                    || 120;
-                this.serverless.cli.log(`Waiting ${waitFor} seconds before starting deployment
-                    because first time creating domain`);
-
-                await sleep(waitFor);
+            const atLeastOneDoesNotExist = () => this.domains.some((domain) => !domain.domainInfo);
+            const maxWaitFor = parseInt(this.serverless.service.custom.customDomain.autoDomainWaitFor, 10) || 120;
+            const pollInterval = 3;
+            for (let i = 0; i * pollInterval < maxWaitFor && atLeastOneDoesNotExist() === true; i++) {
```

Comment:
How do we exit this for loop?

Reply:
if all `domains` exist , then `atLeastOneDoesNotExist()` will be false and exit (`await this.getDomainInfo();` in the for loop will repopulate data so it can go from true/false)

otherwise, if it never populates, the `i * pollInterval` will eventually be greater than `maxWaitFor` and it'll exit and the deployment will eventually fail in the same way as if you never ran `create_domain`

Reply:
@jconstance-amplify let me know what you think :)

Reply:
@jconstance-amplify bump :)

Reply:
@jconstance-amplify no vacations allowed!!

Reply:
Fortunately for you I came back on Monday! :D